### PR TITLE
feat: event restrictions

### PR DESCRIPTION
**Pull request by ipeglin**
## Changes
* Events now can have programme and year restrictions
* Restrictions can only be applied to events with capacity, but are optional
* Fix `setInterval` for SSR in `/event/[id].vue`

### CODE REVIEW

**Review by AasmundN — CHANGES_REQUESTED** [review]
When the user does not fulfil the requirements for registering there is nothing telling the user why it can't register.

Screenshot 2024-02-02 at 17 17 47

**Comment by AasmundN** [comment]
Also, with the same event having only a restriction set to third year students I cannot sign up with a third year basic account.

File: server/api/event/register/index.post.ts [hunk 1]
```diff
@@ -71,6 +71,23 @@ export default defineEventHandler(async (event) => {
       })
     })
 
+  // check if user meets registration requirements
+  const meetsProgrammeRequirement =
+    data[eventUID]?.registration?.requirements?.programmes?.includes(
+      user?.studyProgram
+    ) ?? true
+  const meetsYearRequirement =
+    data[eventUID]?.registration?.requirements?.years?.includes(user?.year) ??
```
Comment (AasmundN) [inline server/api/event/register/index.post.ts:80]: The year key is called `currentYear`, not `year`. That is why it is not working.

**Review by AasmundN — APPROVED** [review]

**Review by AasmundN — DISMISSED** [review]

File: server/api/event/register/index.post.ts [hunk 2]
```diff
@@ -71,6 +71,28 @@ export default defineEventHandler(async (event) => {
...
+    data[eventUID]?.registration?.requirements?.programmes?.includes(
+      user?.studyProgram
+    ) ?? true
+
+  const meetsYearRequirement =
+    data[eventUID]?.registration?.requirements?.years?.includes(
+      user?.currentYear
+    ) ?? true
+
+  console.log(meetsProgrammeRequirement, meetsYearRequirement)
```
Comment (JStoresund) [inline server/api/event/register/index.post.ts:85]: I assume this is for testing. I don't think this should be included in production, even if it is on the server side

**Review by JStoresund — APPROVED** [review]
LGTM. Found one console log which I suggest removing in order to avoid a waterfall of meaningless messages on the server console, but otherwise looks good

File: server/api/event/register/index.post.ts (hunk 2 above)
Comment (AasmundN) [reply server/api/event/register/index.post.ts:85]: Good catch. I'll remove it.

**Review by JStoresund — APPROVED** [review]

**Pull request merged**